Druid/DRCogitCanonicaliser.class.st

```diff
@@ -103,7 +103,11 @@ DRCogitCanonicaliser >> visitBitXor: aDRBitXor [
 { #category : #visiting }
 DRCogitCanonicaliser >> visitBranchIfCondition: aDRConditionalBranch [
 
-	"Nothing"
+	aDRConditionalBranch operands
+		detect: [ :op | op isPhiFunction ]
+		ifFound: [ :thePhi |
+			(thePhi operands allSatisfy: [ :op | op simpleConstantFold isSetConditionCode ])
+				ifTrue: [ thePhi basicBlock tailDuplicate ] ]
```

Comment:
Maybe for a follow-up, I don't know if tail duplicating here wouldn't bring any problems.
The canonicaliser is made to do peephole optimisations, only simplifying locally small sequences of instructions.
Because of this assumption, it iterates the instructions from "top to bottom".

However, tail duplication will duplicate instructions during the iteration.
This is not a problem right now because we make a copy of the instructions before iterating.
But this means that the new copies of the instructions will not be iterated in this pass of the canonicaliser.

Reply:
In other words, my question is wether to move this code to a separate pass :)